src/Symfony/Component/HttpClient/Chunk/ErrorChunk.php

```diff
@@ -101,8 +90,13 @@ public function didThrow(): bool
     public function __destruct()
     {
         if (!$this->didThrow) {
-            $this->didThrow = true;
-            throw new TransportException($this->errorMessage, 0, $this->error);
+            $this->throw();
         }
     }
+
+    private function throw(): void
+    {
+        $this->didThrow = true;
+        throw $this->error instanceof TransportException ? $this->error : new TransportException($this->error->getMessage(), 0, $this->error);
```

Comment:
throwing the original error breaks the trace of the thrown exception, making it harder to understand what's going on